include/seqan3/alignment/matrix/detail/affine_cell_proxy.hpp

```diff
@@ -0,0 +1,108 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides seqan3::detail::affine_cell_proxy.
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <tuple>
+
+#include <seqan3/core/tuple_utility.hpp>
+
+namespace seqan3::detail
+{
+
+/*!\brief A wrapper for an affine score matrix cell.
+ * \implements seqan3::tuple_like
+ * \ingroup pairwise_alignment
+ *
+ * \details
+ *
+ * This wrapper provides a uniform access to the different elements of the cell within an affine score matrix. This includes
+ * the optimal score, the horizontal gap score and the vertical gap score.
+ */
+template <tuple_like tuple_t>
```

Comment:
Can you explain the rationale to me why this is not a fixed tuple type?

Reply:
Because it will be a proxy around the type that is returned from the zip view or the algorithm, which are different tuple types. So it just behaves like the type it inherits from also capturing the correct tuple element types but offers in this case additional interfaces to make reading the code and accessing the values more understandable.